dart/lib/src/transport/transport.dart

```diff
@@ -0,0 +1,125 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:sentry/src/utils.dart';
+
+import '../protocol.dart';
+import '../sentry_options.dart';
+import 'body_encoder_browser.dart' if (dart.library.io) 'body_encoder.dart';
+
+typedef BodyEncoder = List<int> Function(
+  Map<String, dynamic> data,
+  Map<String, String> headers, {
+  bool compressPayload,
+});
+
+typedef HeadersBuilder = Map<String, String> Function(String authHeader);
+
+/// A transport is in charge of sending the event to the Sentry server.
+class Transport {
+  final Client httpClient;
+
+  final Dsn _dsn;
+
+  final Sdk sdk;
+
+  final HeadersBuilder headersBuilder;
+
+  final bool compressPayload;
+
+  /// Used by sentry to differentiate browser from io environment
+  final String platform;
+
+  final ClockProvider _clock;
+
+  /// Use for browser stacktrace
+  final String origin;
```

Comment:
most of those fields if not all are already in the options, should it not just take options and read from there?

Reply:
Updated : the transport is now instantiated from a SentryOptions